### DISCUSSION TITLE
setxkbmap not taking effect from i3 config

### DISCUSSION THREAD

**Discussion by kshepherd** [i3/i3#6088] [Q&A]
I have this line in my i3 config: `exec --no-startup-id setxkbmap -option compose:menu` but it doesn't work. 
Running `setxkbmap -option compose:menu` manually sets my compose key up properly. Is there something I'm doing wrong with my attempt at executing that command on i3 startup?

Thanks!

**Comment by andrei-korshikov** [comment]
Hmm… I would say you do it right. I do almost the same, just with exec_always and empty option, i.e.: `exec_always --no-startup-id setxkbmap -option -option compose:menu`. So I can easily reapply xkb settings on i3 restart. It's useful because sometimes (e.g. after some package updates) they are switched to default. I would say something is rewriting your xkb options right after i3 start.

**Comment by YAUIO** [comment]
`exec --no-startup-id setxkbmap -layout us,ru -option grp:caps_toggle`

maybe adapt it to your needs and it should work

remember that it is lost on kernel regeneration till reboot, and super shift r reload won't fix it for some reason

**Reply by andrei-korshikov** [reply]
> super shift r reload won't fix it for some reason

That's [by design](https://i3wm.org/docs/userguide.html#_automatically_starting_applications_on_i3_startup). See `exec` vs `exec_always` difference.
